app/controllers/flags_controller.rb

```diff
@@ -2,20 +2,20 @@ class FlagsController < ApplicationController
   before_action :authenticate_user!, :except => [:index, :show]
 
   def create
-    answer = Answer.find(params[:answer_id])
+    answer = find_answer(params[:answer_id])
     if current_user.has_flagged? answer
       flash[:notice] = 'You cannot flag more than once'
-      redirect_to "/terms/#{params[:term_id]}"
+      refresh_term_page(params[:term_id])
     else
       answer.flags.create({user_id: current_user.id})
-      redirect_to "/terms/#{params[:term_id]}"
+      refresh_term_page(params[:term_id])
     end
   end
 
   def destroy
-    answer = Answer.find(params[:answer_id])
-    flag = Flag.find(params[:id])
-    if current_user.has_flagged? answer or current_user.admin?
+    answer = find_answer(params[:answer_id])
+    flag = answer.flags.find(params[:id])
+    if current_user.has_flagged? answer
```

Comment:
we need `or current_user.admin?` so admins can unflag, probably a merge conflict issue!